### PR TITLE
Remove mender-client recipes pre 3.1.x series

### PR DESCRIPTION
**Pull request by lluiscampos**
After MEN-3978, these old client won't work anymore with master of
meta-mender due to the tools rename. See 494f92d.

Use the opportunity to clean up some of the conditional logic in
mender-client.inc meant to be compatible with older clients.

Changelog: Title

### CODE REVIEW

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :lacrosse:

File: meta-mender-core/recipes-mender/mender-client/mender-client.inc [hunk 1]
```diff
@@ -254,40 +254,27 @@ do_install() {
...
+        -C ${B}/src/${GO_IMPORT} \
+        V=1 \
+        prefix=${D} \
+        datadir=${datadir} \
+        install-inventory-local-scripts \
+        ${@bb.utils.contains('PACKAGECONFIG', 'inventory-network-scripts', 'install-inventory-network-scripts', '', d)}
+
+    # install dbus files
+    if ${@bb.utils.contains('PACKAGECONFIG', 'dbus', 'true', 'false', d)}; then
+        # install the D-Bus policy file
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-client/mender-client.inc:267]: This part will soon go away as well, see https://github.com/mendersoftware/meta-mender/commit/fb1159a85c37e713a46ed6f7be9a5e5396871dff

**Review by kacf — APPROVED** [review]

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :trumpet:

**Review by oleorhagen — APPROVED** [review]
Some cleanup!

**Pull request merged**